### ISSUE TITLE
CMake's configuration step fails on Ubuntu 20.04 and 21.10

### ISSUE THREAD

**Issue by widgetii** [libevent/libevent#1321]
Steps to reproduce:

```
$ docker run -it ubuntu:20.04
# apt update
# apt install -y build-essential cmake git
# cmake --version
cmake version 3.16.3
# git clone https://github.com/libevent/libevent
# cd libevent
# cmake -H. -Bbuild -DEVENT__DISABLE_OPENSSL=ON -DEVENT__DISABLE_MBEDTLS=ON
...
-- Configuring incomplete, errors occurred!
See also "/libevent/build/CMakeFiles/CMakeOutput.log".
See also "/libevent/build/CMakeFiles/CMakeError.log".
```

`build/CMakeFiles/CMakeError.log` file contents:
```
Determining if files fcntl.h;inttypes.h;memory.h;signal.h;stdarg.h;stddef.h;stdint.h;stdlib.h;string.h;errno.h;unistd.h;time.h;sys/types.h;sys/stat.h;sys/time.h;sys/param.h;netdb.h;dlfcn.h;arpa/inet.h;poll.h;port.h exist failed with the following output:
Change Dir: /libevent/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_b5925/fast && /usr/bin/make -f CMakeFiles/cmTC_b5925.dir/build.make CMakeFiles/cmTC_b5925.dir/build
make[1]: Entering directory '/libevent/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_b5925.dir/EVENT__HAVE_PORT_H.c.o
/usr/bin/cc -D_GNU_SOURCE  -Wall -Wextra -Wno-unused-parameter -Wstrict-aliasing -Wstrict-prototypes -Wundef -fno-strict-aliasing -Wmissing-prototypes -Winit-self -Wmissing-field-initializers -Wdeclaration-after-statement -Waddress -Wnormalized=id -Woverride-init -Wlogical-op -Wwrite-strings -Wno-unused-function    -o CMakeFiles/cmTC_b5925.dir/EVENT__HAVE_PORT_H.c.o   -c /libevent/build/CMakeFiles/CheckIncludeFiles/EVENT__HAVE_PORT_H.c
/libevent/build/CMakeFiles/CheckIncludeFiles/EVENT__HAVE_PORT_H.c:22:10: fatal error: port.h: No such file or directory
   22 | #include <port.h>
      |          ^~~~~~~~
compilation terminated.
make[1]: *** [CMakeFiles/cmTC_b5925.dir/build.make:66: CMakeFiles/cmTC_b5925.dir/EVENT__HAVE_PORT_H.c.o] Error 1
make[1]: Leaving directory '/libevent/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:121: cmTC_b5925/fast] Error 2
```

It seems that issue is in CMake version because 3.24 from Ubuntu 22.04 works
CMake versions 3.16 (from Ubuntu 20.04) and 3.18 (from Ubuntu 21.10) do not work

**Comment by boryssmejda** [comment]
Hi I managed to reproduce the issue locally and the reason for CMake's error is that
in line 181 of CMakeLists.txt there is REAL_PATH used which has been introduced in CMake 3.19
https://cmake.org/cmake/help/latest/command/file.html#path-conversion
![image](https://user-images.githubusercontent.com/33206513/184500308-15dcd58a-c80f-47a9-b9c5-f3f3d1021b0d.png)

There two options here:
1. You either increase the cmake_minimum_version to version 3.19 at least.
2. We try to find a workaround for this issue with some tools already available in CMake 3.16.

I will try to adjust the code to be CMake v3.16 compliant, if it is impossible I will recommend upgrading the minimum CMake version.

**Comment by azat** [comment]
@boryssmejda thank you!
